### ISSUE TITLE
Preservecoord in reductions that select an element

### ISSUE THREAD

**Issue by jl-wynen** [scipp/scipp#3377]
Some reductions (`max`, `min`, `median`, etc.) select one element along a dimension. It is possible to retain coordinates along that dimension by also selecting the corresponding element. Currently, we simply remove all coords along a reduced dim. But if we changed this, we would get sort of combined `max` and `argmax`, etc. operations. I don't know of a cocnrete use case, it's just something that came to my mind.

E.g.
```python
da = sc.DataArray(
    sc.array(dims=['x'], values=[3, 5, 1, 2]),
    coords={'x': sc.array(dims=['x'], values=[1, 2, 3, 4])}
)
da.max()
```
returns
```python
da = sc.DataArray(
    sc.scalar(5),
    coords={'x': sc.scalar(2)}
)
```

**Comment by SimonHeybrock** [comment]
Would you convert to an unaligned coord, such that you can still to `da - da.max()`?

**Comment by SimonHeybrock** [comment]
Have you thought about the cost? This would probably be significantly more expensive to implement?

**Comment by jl-wynen** [comment]
> Would you convert to an unaligned coord, such that you can still to da - da.max()?

Yes, I'd think so. This is akin to indexing into an element where we also preserve the coord.

> Have you thought about the cost? This would probably be significantly more expensive to implement?

Not in detail. But if we implement it as `argmax`, etc., it shouldn't be much more expensive. It would be something like
```python
i = argmax(da.data, dim)
res = da[dim, i]
```

**Comment by SimonHeybrock** [comment]
One could make it on opt-in keyword arg?

**Comment by jl-wynen** [comment]
Possibly.

I need to correct my initial list of functions where this can be used: `median` may return the average of two elements if there is an even number. So it actually cannot return a coord.
